### PR TITLE
Hub deployment ssl

### PR DESCRIPTION
**Pull request by psantos1113**
Changes for deploying to a docker-compose environment with the Hub.

You can start reviewing.  I cannot merge it until a dependency is released. 
This shouldn't conflict with the UI changes.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -20,7 +20,7 @@ plugins {
     id 'com.moowork.node' version '1.2.0'
 }
 
-version = '0.0.1-SNAPSHOT'
+version = '0.0.1-ALPHA'
```
Comment (psantos1113) [inline build.gradle]: Need to change this back to SNAPSHOT before merging into master.

File: src/main/java/com/blackducksoftware/integration/hub/alert/channel/email/EmailMessagingService.java [hunk 2]
```diff
@@ -79,7 +86,14 @@ public void sendEmailMessage(final EmailTarget emailTarget) throws MessagingExce
...
         // TODO allow the ability to upload the image files or use a URL to an image
-        addTemplateImage(model, contentIdsToFilePaths, EmailProperties.EMAIL_LOGO_IMAGE, System.getProperties().getProperty("user.dir") + "/src/main/resources/email/images/Ducky-80.png");
+        final String imagesDirectory = System.getenv("ALERT_IMAGES_DIR");
+        String imageDirectoryPath;
+        if (StringUtils.isNotBlank(imagesDirectory)) {
+            imageDirectoryPath = imagesDirectory + "/Ducky-80.png";
+        } else {
+            imageDirectoryPath = System.getProperties().getProperty("user.dir") + "/src/main/resources/email/images/Ducky-80.png";
+        }
+        addTemplateImage(model, contentIdsToFilePaths, EmailProperties.EMAIL_LOGO_IMAGE, imageDirectoryPath);
```
Comment (gkillough) [inline src/main/java/com/blackducksoftware/integration/hub/alert/channel/email/EmailMessagingService.java]: If this is working properly, can we remove the TODO on line 88?

File: src/main/java/com/blackducksoftware/integration/hub/alert/channel/hipchat/HipChatChannel.java [hunk 3]
```diff
@@ -135,7 +136,15 @@ public Request createRequest(final ChannelRequestHelper channelRequestHelper, fi
...
-            final ChannelFreemarkerTemplatingService freemarkerTemplatingService = new ChannelFreemarkerTemplatingService(System.getProperties().getProperty("user.dir") + "/src/main/resources/hipchat/templates");
+            final String templatesDirectory = System.getenv("ALERT_TEMPLATES_DIR");
+            String templateDirectoryPath;
+            if (StringUtils.isNotBlank(templatesDirectory)) {
+                templateDirectoryPath = templatesDirectory + "/hipchat";
+            } else {
+                templateDirectoryPath = System.getProperties().getProperty("user.dir") + "/src/main/resources/hipchat/templates";
+            }
+
+            final ChannelFreemarkerTemplatingService freemarkerTemplatingService = new ChannelFreemarkerTemplatingService(templateDirectoryPath);
```
Comment (gkillough) [inline src/main/java/com/blackducksoftware/integration/hub/alert/channel/hipchat/HipChatChannel.java]: If this is working properly, can we remove the TODO on line 138?

File: src/main/java/com/blackducksoftware/integration/hub/alert/StartupManager.java [hunk 4]
```diff
@@ -82,7 +83,12 @@ public void logConfiguration() {
         logger.info("Hub Proxy Host:     {}", globalProperties.getHubProxyHost());
         logger.info("Hub Proxy Port:     {}", globalProperties.getHubProxyPort());
         logger.info("Hub Proxy User:     {}", globalProperties.getHubProxyUsername());
-        logger.info("Hub Proxy Password: **********");
+        final String proxyPasswordLabel = "Hub Proxy Password:";
```
Comment (jamesrichard91) [inline src/main/java/com/blackducksoftware/integration/hub/alert/StartupManager.java]: Should we apply this same logic to Hub API Key?

Comment (gkillough) [reply src/main/java/com/blackducksoftware/integration/hub/alert/StartupManager.java]: I think this was only done because the string was repeated.

Comment (psantos1113) [reply src/main/java/com/blackducksoftware/integration/hub/alert/StartupManager.java]: Yeah I would have repeated the string so I just defined it once

File: src/main/java/com/blackducksoftware/integration/hub/alert/hub/controller/global/GlobalHubConfigController.java [hunk 5]
```diff
@@ -57,26 +57,31 @@ public GlobalHubConfigController(final GlobalHubConfigActions configActions, fin
     }
 
     @Override
+    @PreAuthorize("hasRole('ADMIN')")
```
Comment (gkillough) [inline src/main/java/com/blackducksoftware/integration/hub/alert/hub/controller/global/GlobalHubConfigController.java]: I don't believe the "@PreAuthorize" is necessary for anything except the GET. This class has ConfigController in its hierarchy, which has the proper annotations. If this is not working, we probably have a bug in the all of the global controllers.

File: src/main/java/com/blackducksoftware/integration/hub/alert/web/AuthenticationHandler.java [hunk 6]
```diff
@@ -23,19 +23,21 @@
...
 @EnableWebSecurity
 @Configuration
+@ConditionalOnProperty(name = "blackduck.alert.ssl.enable", havingValue = "false", relaxedNames = false)
 public class AuthenticationHandler extends WebSecurityConfigurerAdapter {
 
     @Override
     protected void configure(final HttpSecurity http) throws Exception {
         http.csrf().disable().authorizeRequests().anyRequest().permitAll().and().logout().logoutSuccessUrl("/");
-        // TODO: Make sure the H2 console is disabled by removing the following line of code.
+        // conditional above ensures that this will not be used if SSL is enabled.
```
Comment (gkillough) [inline src/main/java/com/blackducksoftware/integration/hub/alert/web/AuthenticationHandler.java]: And SSL will be used in all customer deployments?

**Comment by psantos1113** [comment]
Worked with @gkillough to test his concerns.  Found an issue beyond the scope of this pull request.  Going to merge and work on that issue separately.

**Pull request merged**